tests/Feature/Conversion/ConversionFileNamerTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Spatie\Medialibrary\Tests\Feature\Conversion;
+
+use Spatie\Medialibrary\Tests\Support\testfiles\TestConversionFileNamer;
+use Spatie\Medialibrary\Tests\TestCase;
+
+class ConversionFileNamerTest extends TestCase
+{
+    /** @test */
+    public function it_can_use_a_custom_conversion_file_namer()
+    {
+        config()->set('medialibrary.conversion_file_namer', TestConversionFileNamer::class);
```

Comment:
My preference would be to add this `TestConversionFileNamer` within this test file, if it's not used anywhere else. No need to making separate files for testing edge cases.

Another approach would be to use an anonymous class.

Reply:
A little more background information about why I feel this way: you could end up with lots of classes all doing something very specific for one or two tests. I find this adds more confusion over time than anything else. The states package is an example of where this grew too complex: https://github.com/spatie/laravel-model-states/tree/master/tests/Dummy

Reply:
I'm going to leave it like it is for now. I plan on doing a big refactor on the tests after release, and will address this at that time. Thanks for the suggestion.